src/filewatch.erl

```diff
@@ -1,107 +1,134 @@
 -module(filewatch).
--export([start/2, stop/1]).
+-export([start/1,
+         start/2,
+         stop/1]).
+
+-type watch_pair() :: {file:name_all(), term()}.
+-export_type([watch_pair/0]).
+
+-define(DRIVER_NAME, filewatch).
+-define(COOLDOWN_MS, application:get_env(filewatch, cooldown_ms, 3000)).
+-define(CALL_ARG, 1).
 
 -type handle() :: pid().
+-type watch_descriptor() :: integer().
+-type directory() :: file:filename().
 
 -record(state,
         {pid :: pid(),
          port :: port(),
-         map :: #{ {integer(), file:filename()} => any(),
-                   file:filename() => integer()}
+         watch_map :: #{
+           {watch_descriptor(), file:filename()} => any(),
+            directory()                          => watch_descriptor()
```

Comment:
Am I correct in assuming that this map spec indicates that there are two possible key types for this one map which both map to different value types? If so I'm curious why we're not using two maps, one for each key value type?